README
This archive contains the distribution Anki-Import,
version 0.007:

  Anki note generation made easy.

This software is copyright (c) 2018 by Steve Dondley.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v6.012.


Changes
Revision history for Anki-Import

0.030     2018-09-21 16:50:00-04:00 America/New_York

0.029     2018-09-17 07:40:04-04:00 America/New_York

0.028     2018-09-16 23:56:09-04:00 America/New_York

0.027     2018-09-16 09:53:37-04:00 America/New_York

0.026     2018-09-13 22:44:41-04:00 America/New_York

0.025     2018-09-12 12:45:48-04:00 America/New_York

0.024     2018-09-11 23:02:44-04:00 America/New_York

0.023     2018-09-11 18:16:10-04:00 America/New_York

0.022     2018-09-05 03:27:23-04:00 America/New_York

0.021     2018-09-04 14:18:21-04:00 America/New_York

0.020     2018-09-04 14:17:06-04:00 America/New_York

0.019     2018-09-04 05:57:48-04:00 America/New_York

0.018     2018-09-03 02:04:15-04:00 America/New_York

0.017     2018-09-03 00:27:17-04:00 America/New_York

0.016     2018-09-03 00:06:56-04:00 America/New_York
Require latest version of Log::Log4perl shortcuts

0.015     2018-09-02 22:50:02-04:00 America/New_York

0.014     2018-09-02 18:48:31-04:00 America/New_York

0.013     2018-09-02 00:26:14-04:00 America/New_York

0.012     2018-09-01 19:46:59-04:00 America/New_York

0.011     2018-08-31 17:27:27-04:00 America/New_York

0.007     2018-08-31 10:46:29-04:00 America/New_York

0.006     2018-08-31 10:33:46-04:00 America/New_York
First release
0.005     2018-08-31 09:53:34-04:00 America/New_York

0.004     2018-08-31 03:43:40-04:00 America/New_York

0.003     2018-08-31 03:30:01-04:00 America/New_York

0.002     2018-08-31 03:12:38-04:00 America/New_York

0.001     2018-08-31 02:27:31-04:00 America/New_York


t/TestUtils.pm
package t::TestUtils 0.000001;

use vars qw(@ISA @EXPORT);

use Carp;
use Test::More;
use Modern::Perl;
use Data::Dumper;
@ISA = qw(Exporter);
@EXPORT = qw/role_check attrib_check method_check module_check tme tmb tm /;

our $msg = '';

# test message begin
sub tmb {
  my $my_msg ||= shift;
  $msg = $my_msg;
  diag ("=======> Begin $msg <=======");
}

sub tme {
  my $new_msg ||= shift;
  $msg = $new_msg ? $new_msg : $msg;
  diag ("------- End $msg -------");
}

sub tm {
  $msg ||= shift;
  diag ($msg);
}

# Module implementation here
sub method_check {
  my $class = shift;
  for my $method (@_) {
    is (!! $class->can($method), 1, 'can ' . $method);
  }
}

# check that object has needed modules
sub module_check {
  foreach my $module (@_) {

    # convert 'My::Module' to 'My/Module.pm'
    $module =~ s/::/\//g;
    $module .= '.pm';

    ok exists $INC{$module}, "$module module loaded";
  }
}

# check object attributes
sub attrib_check {
  my ($class, $attribs) = @_;

  for my $attrib (keys %$attribs) {
    is $class->meta->has_attribute($attrib), 1, "has attribute: $attrib";
    is $class->meta->get_attribute($attrib)->{isa},
        $attribs->{$attrib}{type},
        "has type $attribs->{$attrib}{type}";
    is $class->meta->get_attribute($attrib)->{is},
        $attribs->{$attrib}{read},
        'is ' . ($attribs->{$attrib}{read} eq 'rw' ? 'rw' : 'ro');
    if ($attribs->{$attrib}{lazy}) {
      is $class->meta->get_attribute($attrib)->{lazy},
          $attribs->{$attrib}{lazy},
          'is ' . ($attribs->{$attrib}{lazy} == 1 ? 'lazy' : 'not lazy');
    }
    if ($attribs->{$attrib}{req}) {
      is $class->meta->get_attribute($attrib)->{required},
          $attribs->{$attrib}{req},
          'is ' . ($attribs->{$attrib}{req} == 1 ? 'required' : 'not required');
    }
  }
}

sub role_check {
  my $s = shift;
  for my $role (@_) {
    is ($s->does($role), 1, 'uses ' . $role);
  }
}

1; # Magic true value required at end of module
__END__

=head1 NAME

TestUtils - [One line description of module's purpose here]


=head1 VERSION

This document describes TestUtils version 0.000001


=head1 SYNOPSIS

    use TestUtils;

=for author to fill in:
    Brief code example(s) here showing commonest usage(s).
    This section will be as far as many users bother reading
    so make it as educational and exeplary as possible.


=head1 DESCRIPTION

=for author to fill in:
    Write a full description of the module and its features here.
    Use subsections (=head2, =head3) as appropriate.


=head1 INTERFACE

=for author to fill in:
    Write a separate section listing the public components of the modules
    interface. These normally consist of either subroutines that may be
    exported, or methods that may be called on objects belonging to the
    classes provided by the module.


=head1 DIAGNOSTICS

=for author to fill in:
    List every single error and warning message that the module can
    generate (even the ones that will "never happen"), with a full
    explanation of each problem, one or more likely causes, and any
    suggested remedies.

=over

=item C<< Error message here, perhaps with %s placeholders >>

[Description of error here]

=item C<< Another error message here >>

[Description of error here]

[Et cetera, et cetera]

=back


=head1 CONFIGURATION AND ENVIRONMENT

=for author to fill in:
    A full explanation of any configuration system(s) used by the
    module, including the names and locations of any configuration
    files, and the meaning of any environment variables or properties
    that can be set. These descriptions must also include details of any
    configuration language used.

TestUtils requires no configuration files or environment variables.


=head1 DEPENDENCIES

=for author to fill in:
    A list of all the other modules that this module relies upon,
    including any restrictions on versions, and an indication whether
    the module is part of the standard Perl distribution, part of the
    module's distribution, or must be installed separately. ]

None.


=head1 INCOMPATIBILITIES

=for author to fill in:
    A list of any modules that this module cannot be used in conjunction
    with. This may be due to name conflicts in the interface, or
    competition for system or program resources, or due to internal
    limitations of Perl (for example, many modules that use source code
    filters are mutually incompatible).

None reported.


=head1 BUGS AND LIMITATIONS

=for author to fill in:
    A list of known problems with the module, together with some
    indication Whether they are likely to be fixed in an upcoming
    release. Also a list of restrictions on the features the module
    does provide: data types that cannot be handled, performance issues
    and the circumstances in which they may arise, practical
    limitations on the size of data sets, special cases that are not
    (yet) handled, etc.

No bugs have been reported.

Please report any bugs or feature requests to
C<bug-testutils@rt.cpan.org>, or through the web interface at
L<http://rt.cpan.org>.


=head1 AUTHOR

AUTHOR NAME  C<< <AUTHORID@cpan.org> >>


=head1 LICENCE AND COPYRIGHT

Copyright (c) 2017, AUTHOR NAME C<< <AUTHORID@cpan.org> >>. All rights reserved.

This module is free software; you can redistribute it and/or
modify it under the same terms as Perl itself. See L<perlartistic>.


=head1 DISCLAIMER OF WARRANTY

BECAUSE THIS SOFTWARE IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY
FOR THE SOFTWARE, TO THE EXTENT PERMITTED BY APPLICABLE LAW. EXCEPT WHEN
OTHERWISE STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER PARTIES
PROVIDE THE SOFTWARE "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER
EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE OF THE SOFTWARE IS WITH
YOU. SHOULD THE SOFTWARE PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL
NECESSARY SERVICING, REPAIR, OR CORRECTION.

IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
REDISTRIBUTE THE SOFTWARE AS PERMITTED BY THE ABOVE LICENCE, BE
LIABLE TO YOU FOR DAMAGES, INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL,
OR CONSEQUENTIAL DAMAGES ARISING OUT OF THE USE OR INABILITY TO USE
THE SOFTWARE (INCLUDING BUT NOT LIMITED TO LOSS OF DATA OR DATA BEING
RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD PARTIES OR A
FAILURE OF THE SOFTWARE TO OPERATE WITH ANY OTHER SOFTWARE), EVEN IF
SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF
SUCH DAMAGES.



.build/B0bA7Iog5M/Anki-Import-0.022/Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.012.
use strict;
use warnings;



use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "Anki note generation made easy.",
  "AUTHOR" => "Steve Dondley <s\@dondley.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir::Install" => "0.06"
  },
  "DISTNAME" => "Anki-Import",
  "EXE_FILES" => [
    "bin/anki_import"
  ],
  "LICENSE" => "perl",
  "NAME" => "Anki::Import",
  "PREREQ_PM" => {
    "Cwd" => 0,
    "Exporter" => 0,
    "Getopt::Args" => 0,
    "Log::Log4perl::Shortcuts" => "0.011",
    "Path::Tiny" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Data::Dumper" => 0,
    "File::Path" => 0,
    "Test::Exception" => 0,
    "Test::More" => 0,
    "Test::Output" => 0,
    "Test::Warnings" => 0
  },
  "VERSION" => "0.022",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Cwd" => 0,
  "Data::Dumper" => 0,
  "Exporter" => 0,
  "File::Path" => 0,
  "Getopt::Args" => 0,
  "Log::Log4perl::Shortcuts" => "0.011",
  "Path::Tiny" => 0,
  "Test::Exception" => 0,
  "Test::More" => 0,
  "Test::Output" => 0,
  "Test::Warnings" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}
